dhis-2/dhis-support/dhis-support-expression-parser/src/main/java/org/hisp/dhis/parser/expression/function/FunctionIs.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2004-2022, University of Oslo
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ * Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * Neither the name of the HISP project nor the names of its contributors may
+ * be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.parser.expression.function;
+
+import static java.util.stream.Collectors.toList;
+import static org.hisp.dhis.antlr.AntlrParserUtils.compare;
+import static org.hisp.dhis.parser.expression.antlr.ExpressionParser.ExprContext;
+
+import java.util.List;
+
+import org.hisp.dhis.parser.expression.CommonExpressionVisitor;
+import org.hisp.dhis.parser.expression.ExpressionItem;
+
+/**
+ * Function is
+ *
+ * @author Jim Grace
+ */
+public class FunctionIs
+    implements ExpressionItem
+{
+    @Override
+    public Object evaluate( ExprContext ctx, CommonExpressionVisitor visitor )
+    {
+        Object arg0 = visitor.visit( ctx.expr( 0 ) );
+
+        for ( int i = 1; i < ctx.expr().size(); i++ )
+        {
+            Object argN = visitor.visit( ctx.expr( i ) );
+
+            if ( compare( arg0, argN ) == 0 )
+            {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    @Override
+    public Object evaluateAllPaths( ExprContext ctx, CommonExpressionVisitor visitor )
+    {
+        List<Object> values = ctx.expr().stream()
+            .map( visitor::visit )
+            .collect( toList() );
+
+        Object arg0 = values.get( 0 );
+
+        for ( int i = 1; i < values.size(); i++ )
```

Comment:
FYI: you can use streams with `.skip(1)` to find the match if you wanted

Reply:
Cool, I did not know that. Thanks!

Actually I think I might keep this the same even though the for loop is more clunky. I want the `evaluate` method to exit the loop as soon as it finds a match (not just for performance but there may be side-effects in evaluation of the missing value strategy). There may be some benefit in having the `evaluateAllPaths` logic look as similar as possible, although `.skip(1)` could have been used here as you say.